motionplan/motionPlanner.go

```diff
@@ -427,36 +427,45 @@ func CheckPlan(
 	// The solver frame will have had its PTGs filled in the newPlanManager() call, if applicable.
 	relative := len(sf.PTGSolvers()) > 0
 
+	// where ought the robot be on the plan
+	pathPosition := spatialmath.PoseBetweenInverse(errorState, currentPosition)
+
 	if relative {
 		// get pose of robot along the current trajectory it is executing
 		lastPose, err := sf.Transform(currentInputs)
 		if err != nil {
 			return err
 		}
 
-		// where ought the robot be on the plan
-		pathPosition := spatialmath.PoseBetweenInverse(errorState, currentPosition)
-
 		// absolute pose of the previous node we've passed
 		formerRunningPose := spatialmath.PoseBetweenInverse(lastPose, pathPosition)
 
-		// convert planNode's poses to be in absolute corrdinated
+		// convert planNode's poses to be in absolute coordinates
 		if planNodes, err = rectifyTPspacePath(planNodes, sf, formerRunningPose); err != nil {
 			return err
 		}
 	}
+
 	// adjust planNodes by the errorState
+	// this only changes a node's pose and not its inputs
 	planNodes = transformNodes(planNodes, errorState)
 
 	// pre-pend node with current position of robot to planNodes
 	// Note that currentPosition is assumed to have already accounted for the errorState
-	planNodes = append([]node{&basicNode{pose: currentPosition, q: currentInputs}}, planNodes...)
+	if relative {
+		planNodes = append([]node{&basicNode{pose: currentPosition, q: currentInputs}}, planNodes...)
+	} else {
+		// we adjust the current inputs if we are working with
+		currentInputs[0].Value = pathPosition.Point().X
+		currentInputs[1].Value = pathPosition.Point().Y
```

Comment:
> This is going to break if we try it with anything thats not a diff drive base

It is not immediately clear to me how this would be an issue.
This conditional is set up such that the `else` branch only pertains to bases which specify their coordinates and inputs in non-relative terms - this should then encompass diff drive and other types of bases which are non-PTG.